Add project icon

Pankraz76 opened the pull request:
<img width="172" alt="image" src="https://github.com/user-attachments/assets/f11aeca5-8f7c-405a-985c-c7ead7e7c757" />

https://youtrack.jetbrains.com/issue/IJPL-130477/Welcome-screen-only-SVG-can-be-set-as-a-project-icon

https://github.com/quarkusio/quarkus/blob/main/.idea/icon.png

https://github.com/openrewrite/rewrite/blob/main/.idea/icon.png

sdeleuze commented:
Please update your PR to use https://spring.io/img/projects/spring-framework.svg instead.

pinguin3245678 committed:
add project icon

https://youtrack.jetbrains.com/issue/IJPL-130477/Welcome-screen-only-SVG-can-be-set-as-a-project-icon

https://github.com/openrewrite/rewrite/blob/main/.idea/icon.png

Signed-off-by: Vincent Potucek <vincent.potucek@sap.com>

punkratz312 commented:
Thanks for the learning

sdeleuze commented:
I meant, `.idea/icon.svg` is supported, so I am suggesting to use it directly instead of generating a PNG from it.

sdeleuze commented:
Also please use only one `Signed-off-by:` line if possible in the commit (not sure which one should be used).

bclozel commented:
Shouldn't the icon be resized to be a square? See https://github.com/junit-team/junit5/pull/2240/files for an example.

punkratz312 commented:
We could use a sized PNG or an SVG, as SVG is sizeless since it's a vector-based format that scales without losing quality. I'm happy to apply any image you want—just let me know.

sdeleuze commented:
I made various tries, the SVG looks the nicest option to me, it adapts to light or dark teams, scale to the required size, and seems recommended in IDEA doc. Is it fine for you @bclozel ?

sdeleuze commented:
Merged, thanks.

punkratz312 commented:
Awesome, thank you very much.
